File: cs50x2020/pset2/readability/readability.c
// Readability
// index = 0.0588 * L - 0.296 * S - 15.8
// If index > 16 output: "16+"
// If index < 1, output: "Before grade 1"

#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

void check_grade(int ind);

int main(void)
{
    // Prompt user to input text to be graded
    string text = get_string("Text: ");
    
    // Init counts
    int count_letter = 0, count_word = 0, count_sent = 0;
    
    // Go through every char of the input str
    for (int i = 0; i < strlen(text) ; i++)
    {
        // Check and count if it's an alphabetic letter
        if (isalpha(text[i]))
        {
            count_letter++;
        }
        // Count spaces in order to determine word count (3 words has 2 spaces)
        if (text[i] == ' ')
        {
            count_word++;
        }
        // Count sentences
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count_sent++;
        }
    }
    // Average letter per 100 word
    float avg_L = 100 * ((float)count_letter / (count_word + 1));
    // Average sentence per 100 word
    float avg_S = 100 * ((float)count_sent / (count_word + 1));
    
    // Calculation of Coleman-Liau index
    float index = 0.0588 * avg_L - 0.296 * avg_S - 15.8;
    
    // Determine grade from index
    check_grade(round(index));

}

// Function for determining grade level
void check_grade(int ind)
{
    if (ind > 16)
    {
        printf("Grade 16+\n");
    }
    else if (ind < 16 && ind > 1)
    {
        printf("Grade %i\n", ind);
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: cs50x2020/pset6/readability/readability.py
# Readability
from cs50 import get_string

# Get text from user as input
text = get_string("Text: ")

# Initialize word, letter and sentence counts
# Word count must start from one, text must has at lease one word
word_count = 1
letter_count = 0
sentence_count = 0

# Iterate through text char by char
for c in text:
    # Count char as letter if it's alphabetic
    if c.isalpha():
        letter_count += 1
        continue
    # Count spaces, there are (spacecount + 1) words in text
    if c == " ":
        word_count += 1
        continue
    # Count sentences
    if c == "." or c == "!" or c == "?":
        sentence_count += 1
        continue

# Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8,
# L: avg number of letters per 100 words 
# S: avg number of sentences per 100 words
L = (100 * letter_count) / word_count
S = (100 * sentence_count) / word_count

# Calculate Coleman-Liau index for given L and S values
cl_index = round(0.0588 * L - 0.296 * S - 15.8)

# Print desired grade level
if cl_index > 16:
    print("Grade 16+")
elif cl_index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {cl_index}")

